File: src/algorithms/RemoveDuplicatesfromSortedList.cpp
class Solution
{
public:
    ListNode *deleteDuplicates(ListNode *head)
    {
        if (head == NULL || head->next == NULL)
            return head;
        ListNode *p = head, *q = head->next;
        p->next = NULL;

        while (q)
        {

            if (q->val > p->val)
            {
                p->next = q;
                q = q->next;
                p = p->next;
                p->next = NULL;
            }
            else
            {
                ListNode* temp = q;
                q = q->next;
                delete temp;
            }
        }
        return head;

    }
};

File: Algorithms/python/083-RemoveDuplicatesfromSortedList/RemoveDuplicatesfromSortedList.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head is None:
            return None

        p = head.next
        q = head

        while p is not None:
            if p.val == q.val:
                p = p.next
                q.next = p
            else:
                q = q.next
                p = p.next

        return head
